dali/core/mm/async_pool_test.cu

```diff
@@ -163,12 +162,14 @@ void AsyncPoolTest(Pool &pool, vector<block> &blocks, Mutex &mtx, CUDAStream &st
       CUDA_CALL(cudaMemsetAsync(ptr, fill, size, stream));
       {
         std::lock_guard<Mutex> guard(mtx);
+        (void)guard;  // for dummy mutexes
```

Comment:
Could you explain this? I'm not sure I understand this line correctly

Reply:
Sometimes (in the tests) the mutex is a dummy object where the lock/unlock functions are empty and the compiler is smart enough to see that the construction/destruction has no effect and emits a warning.